src/detect-engine-prefilter.c

```diff
@@ -763,6 +763,59 @@ int PrefilterGenericMpmRegister(DetectEngineCtx *de_ctx, SigGroupHead *sgh, MpmC
     return r;
 }
 
+static void PrefilterMultiGenericMpmFree(void *ptr)
+{
+    // PrefilterMpmListId
+    SCFree(ptr);
+}
+
+static void PrefilterMultiMpm(DetectEngineThreadCtx *det_ctx, const void *pectx, Packet *p, Flow *f,
+        void *txv, const uint64_t idx, const AppLayerTxData *_txd, const uint8_t flags)
+{
+    SCEnter();
+
+    const PrefilterMpmListId *ctx = (const PrefilterMpmListId *)pectx;
+    const MpmCtx *mpm_ctx = ctx->mpm_ctx;
+    SCLogDebug("running on list %d", ctx->list_id);
+    uint32_t local_id = 0;
+
+    while (1) {
```

Comment:
nit: not sure if the compiler optimizes this, but do `do { } while(1)` loop is supposed to be fast if we always enter the loop, saves one loop condition check.

Reply:
nice